GenderPayGap.WebUI/Views/Viewing/EmployerDetails/Employer.cshtml

```diff
@@ -87,102 +85,157 @@
+                                                    ReportStatusBadgeType reportStatus = ReportStatusBadgeType.NotRequired;
+
+                                                    if(report.IsVoluntarySubmission())
                                                     {
-                                                        DateTime deadline = ReportingYearsHelper.GetDeadlineForAccountingDate(report.AccountingDate);
-                                                        <span>
-                                                            This employer
-                                                            @(deadline.AddDays(1) > VirtualDateTime.Now ? "is" : "was")
-                                                            required to report by @(deadline.ToString("d MMMM yyyy"))
-                                                        </span>
+                                                        reportStatus = ReportStatusBadgeType.VoluntarilyReported;
                                                     }
-                                                    else
+                                                    else if (report.IsLateSubmission && report.ReturnId != 0)
                                                     {
-                                                        <span>Not required to report</span>
+                                                        reportStatus = ReportStatusBadgeType.SubmittedLate;
+                                                    }
+                                                    else if(report.IsLateSubmission)
+                                                    {
+                                                        reportStatus = ReportStatusBadgeType.OverDue;
+                                                    }
+                                                    else if(report.ReturnId != 0)
+                                                    {
+                                                        reportStatus = ReportStatusBadgeType.Reported;
+                                                    }
+                                                    else if(Global.ReportingStartYearsToExcludeFromLateFlagEnforcement
+                                                        .Contains(report.AccountingDate.Year)
+                                                         && report.AccountingDate.Year == 2019)
+                                                    {
+                                                        reportStatus = ReportStatusBadgeType.NotRequiredDueToCovid;
+                                                    }
+                                                    else if(organisation.GetIsInscope(report.AccountingDate.Year))
+                                                    {
+                                                        reportStatus = ReportStatusBadgeType.Due;
                                                     }

```

Comment:
Can `ReportStatus` be a property on the object (similar to `OrganisationSize`) rather than something calculated inside the view?